### ISSUE TITLE
Maintain existing file structure - update from 0.2.4 to 0.9.9

### ISSUE THREAD

**Issue by erikanderson** [elastic/puppet-elasticsearch#513]
Our current es cluster is being managed by version 0.2.4 of the plugin. In order to migrate from ES 1.7.1 to 2.1.0 we are updating to the latest version of the plugin. 

Our current data structure looks like this: 
```
/var
  /lib
    /elasticsearch
      /sa-****
        /nodes
          /0
```

Previously we did not name this instance but with the new version of the module I guess we have to. So, I played around with the instance naming a bit. I tried naming the instance to `elasticsearch` but that creates `/var/lib/elasticsearch-elasticsearch`

I thought the following would work: 

``` puppet
  elasticsearch::instance { 'sa-****':
    config        => $config,
    init_defaults => $init_defaults,
    datadir       => '/var/lib/elasticsearch/sa-****',
  }
```

That creates `/var/lib/elasticsearch-sa-****` This would be fine if it is looking at the existing data dir but the process indicates it is looking at `/var/lib/elasticsearch-sa-sand-logyard`:

```
elastic+ 20129 61.2  4.2 4573796 343928 ?      Sl   11:36   0:07 /usr/lib/jvm/java-7-openjdk-amd64//bin/java -Xms2g -Xmx2g -Djava.awt.headless=true -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFr
action=75 -XX:+UseCMSInitiatingOccupancyOnly -XX:+HeapDumpOnOutOfMemoryError -XX:+DisableExplicitGC -Dfile.encoding=UTF-8 -Djna.nosys=true -Des.path.home=/usr/share/elasticsearch -cp /usr/share/elasticsearch/lib/elast
icsearch-2.1.0.jar:/usr/share/elasticsearch/lib/* org.elasticsearch.bootstrap.Elasticsearch start -d -p /var/run/elasticsearch-sa-****.pid -Des.default.path.home=/usr/share/elasticsearch -Des.default.path.logs
=/var/log/elasticsearch/sa-**** -Des.default.path.data=/var/lib/elasticsearch-sa-**** -Des.default.path.work=/tmp/elasticsearch-sa-**** -Des.default.path.conf=/etc/elasticsearch/sa-****
```

Any ideas on how I can maintain my existing file structure with the new plugin?

**Comment by electrical** [comment]
Hmm, when setting the data dir in the instance config it should honour that.
Let me double check tomorrow in my tests and see if there is a bug.

**Comment by erikanderson** [comment]
Thank you, I am also taking a look at the module itself to see if that can be improved

**Comment by quixoten** [comment]
It doesn't look like the data directory set on an instance is making it into [the init.d script](https://github.com/elastic/puppet-elasticsearch/blob/master/templates/etc/init.d/elasticsearch.Debian.erb#L83). The `CONF_DIR` is another example of a variable in the `init.d` script that will not honor what is set on the instance.

**Comment by electrical** [comment]
the init scripts are not changed but the defaults file is at `/etc/sysconfig/elasticsearch-$instancename`
Also the data dir is being set in the elasticsearch.yml file in the instance config dir.

**Comment by electrical** [comment]
Hi,

The test at https://github.com/elastic/puppet-elasticsearch/blob/master/spec/acceptance/005_datapath_spec.rb#L129-L177 tests your specific situation and should work fine.

Can you check the settings in your elasticsearch.yml file?

Also could you check the content of $init_defaults ?  that you set in your instance config ?

Cheers.

**Comment by electrical** [comment]
I did find out where the `-Des.default.path.data` comes from but it should be overridden by the values that are in the elasticsearch.yml file.

**Comment by quixoten** [comment]
I work with @erikanderson here at mx.com and the issue we were having was that we were making the wrong assumptions about how the data path for an instance in puppet translated to the data path that elasticsearch uses. We did not account for elasticsearch using the cluster.name as a sub-directory, so we were ending up with data in `/es/data/instance-name/cluster-name`, but what we wanted is `/es/data/cluster-name`.

Figured out how to get that with your module, so this is a non-issue. Thanks for you help!
